### DISCUSSION TITLE
What's the meaning of HRA's spec.scaleUpTriggers.duration?

### DISCUSSION THREAD

**Discussion by brennerm** [actions/actions-runner-controller#890] [Questions] [answered]
I don't understand the meaning of the `spec.scaleUpTriggers.duration` HRA field and the CRD doesn't contain any info as well. Didn't find any info in the docs as well.

Would appreciate some explanation and I'm happy to improve the docs afterwards.

**Comment by toast-gear** [comment, marked as answer]
The scale up trigger’s duration is used to calculate if the replica number added via the trigger is expired or not. On each reconcilation loop, the controller sums up all the non-expiring replica numbers from previous scale up triggers. It then compares the summed desired replica number against the current replica number. If the summed desired replica number > the current number then it means the replica count needs to scale up.

The thing that has the ultimate say on how long a replica lives however is the `scaleDownDelaySecondsAfterScaleOut` property. If the latest scale-up time + the anti-flapping duration is later than the current time, it doesn’t immediately scale up and instead retries the calculation again later to see if it needs to scale yet.

These are details that is missing really that need added. I'm doing a significant rework of the auto-scaling docs atm and will include details on how the `duration` property is used and how it interacts. https://github.com/actions-runner-controller/actions-runner-controller/tree/docs/auto-scaling-cleanup

**Reply by brennerm** [reply]
Thanks, that explains it very well. :+1:
